Palindrome products:  sync expected test results and input data with problem-specifications.

BethanyG opened the pull request:
Part of https://github.com/exercism/python/issues/1762

- Changed function names to **largest** /  **smallest** in tests, `example.py`, and exercise stub to conform with `canonical-data.json`.
- Changed main function function argument names to **min**, / **max** in tests, `example.py`, and exercise stub to conform with `canonical-data.json`.
- Changed expected data type in tests to **list of lists** to conform with `canoncial-data.json`.
- Changed some function and variable names in `example.py` to better clarify code and remove single letter variable names.
- ~Removed `from __future__ import division` from `example.py`~

cmccandless commented on exercises/palindrome-products/example.py:
```diff
@@ -1,53 +1,51 @@
-def largest_palindrome(max_factor, min_factor):
-    return get_extreme_palindrome_with_factors(max_factor, min_factor,
-                                               "largest")
+def largest(min, max):
```
`min` and `max` are reserved.

```suggestion
def largest(min_factor, max_factor):
```

cmccandless commented on exercises/palindrome-products/palindrome_products_test.py:
```diff
@@ -1,10 +1,10 @@
 """
 Notes regarding the implementation of smallest_palindrome and
```
```suggestion
Notes regarding the implementation of smallest and
```

BethanyG commented:
Since I needed to alter the tests as well, decided to push additional changes.

*  Changed *min/max* back to *min_factor / max_factor in tests, test notes, example code, and exercise stub to avoid using *min/mx* reserved words for argument names.
*  Changed reference to smallest_pallendrome to *smallest*  in test notes.

cmccandless commented:
Merged; thanks for working on this!
